Makefile

```diff
@@ -135,6 +198,26 @@ PKG_CONFIG := pkg-config
 endif
 endif
 
+# Libtool makes it easy to correctly build shared libraries with
+# version info on both MacOS and GNU/Linux; require it if building
+# libraries.
+ifneq ($(LIBRARY),)
+ifneq (, $(shell command -v libtool 2> $(NULL)))
+LIBTOOL := libtool
+LIBTOOL_CC := $(LIBTOOL) --tag=CC --mode=compile $(CC) -c
+LIBTOOL_LD := $(LIBTOOL) --tag=CC --mode=link $(CC) \
```

Comment:
And you really shouldn't be using libtool at all if you aren't use autotools, this is not going to be portable for slibtool.

Reply:
Seems slibtool (which I had no idea existed until now) is not fully compatible with libtool; am I not sure I understand the compatibility argument. Is it not a fault of slibtool to not being fully compatible with libtool? Where would this be an issue?

Reply:
Gentoo allows using slibtool where it works with most packages.

https://wiki.gentoo.org/wiki/Slibtool

Using libtool outside of autotools is always problematic and an extremely suboptimal way to use libtool. The GNU libtool script is one of the worst things ever written (Which is why slibtool exists).

Reply:
> Using libtool outside of autotools is always problematic and an extremely suboptimal way to use libtool.

Could you expound on why you say it's "always problematic" to use libtool outside of Autotools? I'm curious of the reasons. I've found some peculiarities a bit challenging while authoring this diff, but I don't foresee problems with it. If Gentoo prefers slibtool, it seems that should work should work using LIBTOOL=slibtool as well, assuming the basic features of libtool I've made used are covered by this alternative implementation of it.

Regarding the quality of libtool itself, I don't particularly mind how pretty or ugly the innards of its implementation are; it's a battle-tested solution that simplified the problem at hand.

I don't mind revising this series more, but I'll need specific problems to address (I don't count "libtool is ugly" as one). Thank you for your continued input!

Reply:
GNU libtool is supposed to work by generating a `./libtool` in the build tree during the `./configure` process using `LT_INIT` from `autoconf`. The default slibtool symlink to use is `rlibtool` which depends on this existing so it can determine if it needs shared or static libraries and it will fail if this does not exist or if `slibtoolize` is not used instead.

Since this is not autotools there is `./configure`, `LT_INIT` or `libtoolize`/`slibtoolize` and it will fail with flow errors.

https://wiki.gentoo.org/wiki/Slibtool#flow_error:_unexpected_condition_or_other

To be very blunt, this solution is entirely a non-starter.

Reply:
Could we s/libtool/slibtool/ and be done, if it fixes all these libtool shortcomings? Or is more complicated than than to switch to slibtool?